src/opm/input/eclipse/Schedule/MSW/WellSegments.cpp

```diff
@@ -349,52 +443,65 @@ namespace Opm {
+    void WellSegments::processINC(double depth_top, double length_top)
+    {
+        // Update the information inside the WellSegments to be in ABS way
         Segment new_top_segment(this->m_segments[0], depth_top, length_top);
         this->addSegment(new_top_segment);
+
         orderSegments();
 
-        // begin with the second segment
-        for (std::size_t i_index= 1; i_index< size(); ++i_index) {
-            if( m_segments[i_index].dataReady() ) continue;
+        // Begin with the second segment
+        for (std::size_t i_index = 1; i_index < size(); ++i_index) {
+            if (m_segments[i_index].dataReady()) {
+                continue;
+            }
 
-            // find its outlet segment
+            // Find its outlet segment
             const int outlet_segment = m_segments[i_index].outletSegment();
-            const int outlet_index= segmentNumberToIndex(outlet_segment);
+            const int outlet_index = segmentNumberToIndex(outlet_segment);
 
             // assert some information of the outlet_segment
-            assert(outlet_index>= 0);
+            assert(outlet_index >= 0);
             assert(m_segments[outlet_index].dataReady());
 
             const double outlet_depth = m_segments[outlet_index].depth();
             const double outlet_length = m_segments[outlet_index].totalLength();
             const double temp_depth = outlet_depth + m_segments[i_index].depth();
             const double temp_length = outlet_length + m_segments[i_index].totalLength();
-
-            // applying the calculated length and depth to the current segment
-            Segment new_segment(this->m_segments[i_index], temp_depth, temp_length);
-            addSegment(new_segment);
+            const auto new_x = m_segments[outlet_index].node_X() + m_segments[i_index].node_X();
+            const auto new_y = m_segments[outlet_index].node_Y() + m_segments[i_index].node_Y();
+
+            // Applying the calculated length and depth to the current segment
+            this->addSegment(Segment {
+                this->m_segments[i_index],
+                temp_depth, temp_length, new_x, new_y
+            });
         }
     }
 
-    void WellSegments::orderSegments() {
-        // re-ordering the segments to make later use easier.
-        // two principles
-        // 1. the index of the outlet segment will be stored in the lower index than the segment.
-        // 2. the segments belong to the same branch will be continuously stored.
-
-        // top segment will always be the first one
-        // before this index, the reordering is done.
-        std::size_t current_index= 1;
-
-        // clear the mapping from segment number to store index
+    void WellSegments::orderSegments()
+    {
+        // Re-ordering the segments to make later use easier.
+        //
+        // Two principles
+        //   1. The index of the outlet segment will be stored in the lower
+        //      index than the segment.
```

Comment:
Not sure what you mean by this.

Reply:
That a segment's outlet segment is ordered before the segment itself.  I've pushed an update to clarify.